pkg/apiserver/webhook/configmap_webhook.go

```diff
@@ -106,24 +110,28 @@ func (v *ConfigMapValidator) validateUpdate(req admission.Request) admission.Res
 		return admission.Errored(http.StatusBadRequest, err)
 	}
 
-	oldCloudResourcePrefix, err := v.getCloudResourcePrefix(oldConfigMap)
+	oldControllerConfig, err := v.getControllerConfig(oldConfigMap)
 	if err != nil {
 		return admission.Errored(http.StatusBadRequest, err)
 	}
 
-	if newCloudResourcePrefix == oldCloudResourcePrefix {
-		return admission.Allowed("")
-	}
-	if v.NpControllerInterface.IsCloudResourceCreated() {
-		return admission.Denied(fmt.Sprintf("cloud resource is already created with CloudResourcePrefix %s. "+
-			"Please delete the resource and try or use the same CloudResourcePrefix",
-			oldCloudResourcePrefix))
+	if newControllerConfig.CloudResourcePrefix != oldControllerConfig.CloudResourcePrefix {
+		if v.NpControllerInterface.IsCloudResourceCreated() {
+			return admission.Denied(fmt.Sprintf("cloud resource is already created with CloudResourcePrefix %s. "+
+				"Please delete the resource and try or use the same CloudResourcePrefix", oldControllerConfig.CloudResourcePrefix))
```

Comment:
nit: Please delete the resources and retry. Drop 'or same prefix'

Reply:
```suggestion
				"Please delete the resource(s) and retry.
```

Reply:
Solved